# IS 362 - Project 2

## Objective:
The goal of this assignment is to practice tidying, transforming, and analyzing datasets. In this project, I will be using three datasets:
- Metacritic scores of video games
- Popularity metrics of video games
- Video game sales data

I will read in these datasets, clean the data, transform as necessary, and perform some basic analysis.

## Step 1: Loading the Datasets

I will start by loading the three wide datasets using pandas and preview the first few rows of each dataset to understand their structure.

In [20]:
import pandas as pd

gamesmetacritic = pd.read_csv(r"C:\Users\The King\Desktop\FALL 2024\IS 362\assignment week 5\is362_Project2\gamesmetacriticsscore.csv")
gamespopularity = pd.read_csv(r"C:\Users\The King\Desktop\FALL 2024\IS 362\assignment week 5\is362_Project2\gamespopularity.csv")
vgsales = pd.read_csv(r"C:\Users\The King\Desktop\FALL 2024\IS 362\assignment week 5\is362_Project2\vgsales.csv")

print(gamesmetacritic.head())
print(gamespopularity.head())
print(vgsales.head())

                                           name platform       release_date  \
0                            ! SPACE ACCIDENT !       PC  February 10, 2022   
1  ! That Bastard Is Trying To Steal Our Gold !       PC       May 11, 2014   
2                                    !4RC4N01D!       PC   January 12, 2018   
3                   !4RC4N01D! 2: Retro Edition       PC   February 6, 2018   
4                      !4RC4N01D! 3: Cold Space       PC      March 8, 2018   

                                             summary metascore userscore  
0  The year is 2119. A turning point has begun in...       tbd       tbd  
1  It's a fun puzzle game where you need to steal...       tbd       3.4  
2  Hardcore arkanoid in the spirit of old games f...       tbd       4.0  
3  Everyone dreams of returning 2007, but no one ...       tbd       3.8  
4  !4RC4N01D! returns! This time we have an accid...       tbd       3.0  
   Unnamed: 0                                    Title  Release Date  \
0  

In [19]:
import pandas as pd

non_numeric_rows = tidy_popularity[~tidy_popularity['Popularity_Score'].apply(lambda x: pd.api.types.is_numeric_dtype(type(x)))]

print("Non-numeric Popularity_Score entries:")
print(non_numeric_rows)

Non-numeric Popularity_Score entries:
                                         Title Popularity_Metric  \
1512                                Elden Ring      Release Date   
1513                                     Hades      Release Date   
1514   The Legend of Zelda: Breath of the Wild      Release Date   
1515                                 Undertale      Release Date   
1516                             Hollow Knight      Release Date   
...                                        ...               ...   
13603             Back to the Future: The Game           Reviews   
13604                        Team Sonic Racing           Reviews   
13605                           Dragon's Dogma           Reviews   
13606                          Baldur's Gate 3           Reviews   
13607                 The LEGO Movie Videogame           Reviews   

                                        Popularity_Score  
1512                                        Feb 25, 2022  
1513                       

## Step 2: Exploring and Understanding the Data

Once the datasets are loaded, I will explore their structure, check for missing values, and review basic statistics to better understand the data before tidying it.

In [21]:
print(gamesmetacritic.info())
print(gamespopularity.info())
print(vgsales.info())

print(gamesmetacritic.isnull().sum())
print(gamespopularity.isnull().sum())
print(vgsales.isnull().sum())

print(gamesmetacritic.describe())
print(gamespopularity.describe())
print(vgsales.describe())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 142417 entries, 0 to 142416
Data columns (total 6 columns):
 #   Column        Non-Null Count   Dtype 
---  ------        --------------   ----- 
 0   name          142417 non-null  object
 1   platform      142417 non-null  object
 2   release_date  142417 non-null  object
 3   summary       124702 non-null  object
 4   metascore     142417 non-null  object
 5   userscore     142417 non-null  object
dtypes: object(6)
memory usage: 6.5+ MB
None
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1512 entries, 0 to 1511
Data columns (total 14 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Unnamed: 0         1512 non-null   int64  
 1   Title              1512 non-null   object 
 2   Release Date       1512 non-null   object 
 3   Team               1511 non-null   object 
 4   Rating             1499 non-null   float64
 5   Times Listed       1512 non-null   object 
 6   Numbe

## Step 3: Tidying the Data

These datasets are in wide format, meaning that there are multiple columns representing different variables. To make them easier to analyze, I will use `pandas.melt()` to convert them into tidy format, where each variable forms a column and each observation forms a row.

In [27]:
tidy_metacritic = pd.melt(gamesmetacritic, 
                          id_vars=["name", "platform", "release_date"],
                          value_vars=["metascore", "userscore"], 
                          var_name="Score_Type", 
                          value_name="Score")

tidy_metacritic['Score'] = pd.to_numeric(tidy_metacritic['Score'], errors='coerce')
tidy_metacritic_clean = tidy_metacritic.dropna(subset=['Score'])
print(tidy_metacritic_clean.head())

tidy_popularity = pd.melt(gamespopularity,
                          id_vars=["Title", "Release Date", "Rating"],
                          value_vars=["Plays", "Playing", "Backlogs", "Wishlist"],
                          var_name="Popularity_Metric",
                          value_name="Popularity_Score")

# Convert 'Popularity_Score' to numeric and handle errors
tidy_popularity['Popularity_Score'] = pd.to_numeric(tidy_popularity['Popularity_Score'], errors='coerce')
tidy_popularity_clean = tidy_popularity.dropna(subset=['Popularity_Score'])
print(tidy_popularity_clean.head())

tidy_vgsales = pd.melt(vgsales,
                       id_vars=["Name", "Platform", "Year", "Genre"],
                       value_vars=["NA_Sales", "EU_Sales", "JP_Sales", "Other_Sales", "Global_Sales"],
                       var_name="Region", 
                       value_name="Sales")

tidy_vgsales_clean = tidy_vgsales.dropna(subset=['Sales'])
print(tidy_vgsales_clean.head())


               name       platform       release_date Score_Type  Score
25         "klocki"            iOS      July 13, 2016  metascore   83.0
32           #DRIVE         Switch  February 16, 2021  metascore   69.0
39           #IDARB       Xbox One   January 30, 2015  metascore   77.0
40  #KILLALLZOMBIES       Xbox One    August 10, 2016  metascore   53.0
43  #KILLALLZOMBIES  PlayStation 4   October 28, 2014  metascore   63.0
                          Title  Release Date  Rating Popularity_Metric  \
219              Soul Hackers 2  Aug 25, 2022     3.1             Plays   
274       Flower, Sun, and Rain  May 02, 2001     4.3             Plays   
318             Digimon Survive  Jul 28, 2022     3.4             Plays   
486              Soul Hackers 2  Aug 25, 2022     3.1             Plays   
514  Full Metal Daemon Muramasa  Oct 30, 2009     4.3             Plays   

     Popularity_Score  
219             750.0  
274             632.0  
318             454.0  
486             750.0

## Step 4: Analyzing the Tidied Datasets

After tidying the datasets, I performed analyses to gain insights into **Metacritic scores**, **popularity metrics**, and **regional sales**. In this step, I calculated average values for each of these categories to identify trends and patterns in game performance, reception, and popularity.

### Analyzing Metacritic Scores
I separated the Metascore and Userscore metrics and calculated the average scores for each game, platform, and release date combination. The average Metascore reflects the critical reception, while the Userscore reflects the reception from players.

### Analyzing Popularity Metrics
I calculated average popularity metrics such as Plays, Backlogs, and Wishlists. These metrics provide insight into how frequently games are played, added to backlogs, or desired by players.

Additionally, I identified the top 5 games by average popularity score to showcase the most popular titles.

### Analyzing Video Game Sales by Region
I analyzed video game sales by region, grouping sales by game name, platform, genre, and region. This helps understand the geographical distribution of game sales, highlighting where games performed well globally, including regions like North America, Europe, and Japan.

In [29]:
metascores = tidy_metacritic_clean[tidy_metacritic_clean["Score_Type"] == "metascore"]
userscores = tidy_metacritic_clean[tidy_metacritic_clean["Score_Type"] == "userscore"]

average_metascores = metascores.groupby(["name", "platform", "release_date"])["Score"].mean().reset_index()
print("Average Metascores by Game:")
print(average_metascores.head())

average_userscores = userscores.groupby(["name", "platform", "release_date"])["Score"].mean().reset_index()
print("Average Userscores by Game:")
print(average_userscores.head())

average_popularity = tidy_popularity_clean.groupby("Popularity_Metric")["Popularity_Score"].mean().reset_index()
print("Average Popularity Metrics:")
print(average_popularity)

top_popularity = tidy_popularity_clean.groupby("Title")["Popularity_Score"].mean().reset_index().sort_values(by="Popularity_Score", ascending=False).head(5)
print("Top 5 Games by Popularity:")
print(top_popularity)

total_sales = tidy_vgsales_clean.groupby(["Name", "Platform", "Genre", "Region"])["Sales"].sum().reset_index()
print("Total Sales by Region:")
print(total_sales.head())

Average Metascores by Game:
              name       platform       release_date  Score
0         "klocki"            iOS      July 13, 2016   83.0
1           #DRIVE         Switch  February 16, 2021   69.0
2           #IDARB       Xbox One   January 30, 2015   77.0
3  #KILLALLZOMBIES  PlayStation 4   October 28, 2014   63.0
4  #KILLALLZOMBIES       Xbox One    August 10, 2016   53.0
Average Userscores by Game:
                                           name platform      release_date  \
0  ! That Bastard Is Trying To Steal Our Gold !       PC      May 11, 2014   
1                                    !4RC4N01D!       PC  January 12, 2018   
2                   !4RC4N01D! 2: Retro Edition       PC  February 6, 2018   
3                      !4RC4N01D! 3: Cold Space       PC     March 8, 2018   
4                 !4RC4N01D! 4: KOHBEEP edition       PC    March 30, 2018   

   Score  
0    3.4  
1    4.0  
2    3.8  
3    3.0  
4    3.0  
Average Popularity Metrics:
  Popularity_Metric  

## Step 5: Conclusions and Insights

From the analysis performed on the video game datasets, several key trends and insights emerged:

### Metacritic Scores:
- **Critical Reception**: The game **"klocki"** on iOS received a high Metascore of **83.0**, indicating a strong reception from critics. Similarly, games like **"#DRIVE"** and **"!KILLALLZOMBIES"** also performed relatively well with Metascores above 60.
- **Player Reception**: Interestingly, the game **"I That Bastard Is Trying To Steal Our Gold!"** on PC received a range of userscores averaging around **3.4**, suggesting that player reception was more divided. This indicates that player preferences can sometimes differ significantly from critical reviews.

### Popularity Metrics:
- **Highly Popular Games**: Based on the analysis, **"It Takes Two"** stands out as the most popular game in terms of user interaction, with a **Popularity Score of 999.0**. Other popular games include **"Kirby and the Forgotten Land"**, **"Death Stranding"**, and **"Sekiro: Shadows Die Twice"**. These titles have high interaction scores, indicating strong player engagement and sustained interest.
- **Engagement Insights**: High scores in "Plays" and "Backlogs" suggest that players are highly engaged with these games, often adding them to their wishlists or planning to play them in the future. This is especially true for popular games that offer replayability or unique mechanics, such as **It Takes Two**.

### Video Game Sales:
- **Regional Success**: Sales data highlights the dominance of **'98 Koshien** on PlayStation in the **European**, **Japanese**, and **Global** markets. This indicates that sports games, particularly in the Japanese region, have strong regional popularity.
- **Platform Trends**: Certain platforms such as **PlayStation** show strong regional sales, especially for genres like sports games, which performed better in both **Europe** and **Japan** than other genres.